languages/rust/oso/src/oso.rs

```diff
@@ -131,9 +138,11 @@ impl Oso {
     }
 
     /// Clear out all files and rules that have been loaded.
-    pub fn clear_rules(&self) {
+    pub fn clear_rules(&mut self) -> crate::Result<()> {
```

Comment:
I'm pretty sure these methods take a `&self` so that the `Oso` object can be shared across threads. Why did we need to change them?

Reply:
They are initialization methods, and other initialization methods like `register_class` take `&mut self`, so this might be okay if we can't avoid it.

Reply:
`enable_roles` sets a flag on the `Oso` object so any method that might call it needs to have a mutable reference.